src/core/vector/qgsvectorlayer.cpp

```diff
@@ -3466,6 +3444,9 @@ QgsFeatureSource::FeatureAvailability QgsVectorLayer::hasFeatures() const
 
 bool QgsVectorLayer::commitChanges( bool stopEditing )
 {
+  if ( QgsProject::instance()->transactionMode() == Qgis::TransactionMode::BufferedGroups )
+    return QgsProject::instance()->commitChanges( mCommitErrors, stopEditing, this );
```

Comment:
Best to use it consistently throughout the file

Reply:
I have changed it everywhere i used it for this PR. In numerous other places `QgsVectorLayer` uses `QgsProject::instance()`. I think all this occurrences should also be updated... I could do it in a separate pr

Reply:
Yes, please do, IIRC I cleaned that up a year ago but there is no way to check that in a future proof way.